### PR TITLE
Servicemodulefix

### PR DESCRIPTION
**Pull request by Pixelnator**
- Ported over the RSF (Rapid Service Fabricator) from old Unity. (Refactored code and allows borgs to print items into peoples hands similar to the "give" verb)
- Gave borgs a verb to toggle their own cover lock. Handy if you want someone to switch your battery/change your module but they lack access.
- Changed the languages the service module gives you to facilitate the updated languages of the server

The adjacency check on printing items into peoples hands doesn't work and I can't figure out how to fix it. If someone else wants to do it, feel free.

### CODE REVIEW

File: code/game/objects/items/weapons/RSF.dm [hunk 1]
```diff
@@ -1,116 +1,366 @@
...
+				return 0
+
+	attack(mob/M as mob, mob/user as mob)
+		if (istype(M, /mob/living/carbon/human) && M != usr) //Only give items to humanoid characters and only if you're giving it to someone else.
+			if (checkResource(1, user)) //Only give the prompt if the RSF has ammo or the borg has battery
+				switch(alert(M,"[usr] wants to print you \a [setting]?",,"Yes","No"))
+					if("Yes")
+						switch(mode)
+							if (1)
+								if(!Adjacent(usr))
```
Comment (Pixelnator) [inline code/game/objects/items/weapons/RSF.dm:177]: This is the bit that doesn't work. If you can figure out how to make it work, go nuts. It's supposed to check that the two characters are still next to each other but right now it checks if the player is next to itself (I think) and all my attempts to fix it result in always getting false.

**Comment by Jamini** [comment]
Pull Requests go to Dev, excepting hotfixes and map updates.

File: code/game/objects/items/weapons/RSF.dm [hunk 2]
```diff
@@ -1,116 +1,366 @@
...
+//Updated by Pixelnator on 12.8.2015 to work with Unity2.0 code.
+
 /*
 CONTAINS:
 RSF
-
 */
-
 /obj/item/weapon/rsf
-	name = "\improper Rapid-Service-Fabricator"
```
Comment (Jamini) [inline code/game/objects/items/weapons/RSF.dm:8]: \improper is used for some parts of the naming code, mostly for including "The XXX" and for some forms of areas. Please re-add it

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 3]
```diff
@@ -58,24 +58,35 @@
...
+	R.add_language("Pekhotha sign", 0)
+	R.add_language("Skrellian", 0)
+	R.add_language("Sini", 0)
+	R.add_language("Kida", 0)
 	R.add_language("Rootspeak", 0)
+	R.add_language("Sign language", 0) //yes, the borg knows sign language. If you need to know why it can use it, a space wizard did it.
+	R.add_language("Zho", 0)
+	R.add_language("Ara", 0)
+	R.add_language("Hindi", 0)
+	R.add_language("", 0)
```
Comment (Jamini) [inline code/modules/mob/living/silicon/robot/robot_modules.dm:78]: I have a fix for robot languages staged. Will try a manual merge on those files once this is cleaned up and sent to dev.

**Pull request closed without merging**